lively.components/list.js

```diff
@@ -1870,29 +1873,36 @@ export class DropDownListModel extends ButtonModel {
     this.onRefresh('selection');
   }
 
+  updateListMorphIfNeeded () {
+    const { listMorph: { items: itemsInList }, items } = this;
+    const itemsChanged = !obj.equals(itemsInList.map(m => m.value), items.map(m => m.value));
+    if (itemsChanged) { this.listMorph.items = items; return true; }
+  }
+
   onRefresh (prop) {
     const { listMorph } = this;
-    const sel = this.selection;
 
     if (!listMorph) return super.onRefresh(prop);
 
-    listMorph.items = this.items;
-
-    if (!sel) {
-      listMorph.selection = null;
-      this.label = { value: '' };
-    } else {
-      const { items } = listMorph;
-      let item = listMorph.find(sel);
-      if (!item && typeof sel === 'string') {
-        item = items.find(ea => ea.string === sel);
-      }
-      if (!item) return super.onRefresh(prop);
+    if (this.updateListMorphIfNeeded() || prop === 'selection' || prop === 'items') {
+      const sel = this.selection;
+      if (!sel) {
+        listMorph.selection = null;
+        this.label = { value: '' };
+      } else {
+        const { items } = listMorph;
+        let item = listMorph.find(sel);
+        if (!item && typeof sel === 'string') {
+          item = items.find(ea => ea.string === sel);
+        }
+        if (!item) return super.onRefresh(prop);
 
-      if (prop === 'selection') this.adjustLableFor(item);
+        this.adjustLableFor(item);
```

Comment:
This now done more than previously, was this intended? The condition above is less strong than the old one, no?

Reply:
Slightly so, but really only if the condition `this.updateListMorphIfNeeded() || prop === 'selection' || prop === 'items'` is true, which is not super more often than before.